### PR TITLE
Gh4034 CollectorCode and CustomerCode in Der Kurier

### PR DESCRIPTION
**Pull request by metas-rc**
https://github.com/metasfresh/metasfresh/issues/4034

### CODE REVIEW

File: de.metas.shipper.gateway.derkurier/src/main/java/de/metas/shipper/gateway/derkurier/misc/DerKurierShipperConfig.java [hunk 1]
```diff
@@ -67,5 +72,8 @@ private DerKurierShipperConfig(
 				"Parameter parcelNumberAdSequenceId is > 0");
 		this.parcelNumberAdSequenceId = parcelNumberAdSequenceId;
 		this.parcelNumberGenerator = new ParcelNumberGenerator(parcelNumberAdSequenceId);
+		
+		this.collectorCode = collectorCode;
```
[the remark this answers is not in the document]
Comment (metas-rc) [reply de.metas.shipper.gateway.derkurier/src/main/java/de/metas/shipper/gateway/derkurier/misc/DerKurierShipperConfig.java]: Done. I made it mandatory in both config and deliveryorderline

File: de.metas.shipper.gateway.derkurier/src/main/java/de/metas/shipper/gateway/derkurier/DerKurierDeliveryOrderRepository.java [hunk 2]
```diff
@@ -394,6 +395,9 @@ public DeliveryOrder save(@NonNull final DeliveryOrder deliveryOrder)
 			lineRecord.setDK_ParcelNumber(derKurierDeliveryData.getParcelNumber());
 
 			lineRecord.setDK_Reference(deliveryOrder.getCustomerReference());
+			
+			lineRecord.setCollectorCode(deliveryOrder.getCollectorCode());
```
[the remark this answers is not in the document]
Comment (metas-rc) [reply de.metas.shipper.gateway.derkurier/src/main/java/de/metas/shipper/gateway/derkurier/DerKurierDeliveryOrderRepository.java]: Done

File: de.metas.shipper.gateway.derkurier/src/main/sql/postgresql/system/99-de.metas.shipper.gateway.derkurier/5493040_sys_gh4034_DerKurierShipperConfig_CustomerCode_And_CollectorCode.sql [hunk 3]
```diff
@@ -0,0 +1,52 @@
+-- 2018-05-09T13:19:52.922
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+INSERT INTO AD_Column (AD_Client_ID,AD_Column_ID,AD_Element_ID,AD_Org_ID,AD_Reference_ID,AD_Table_ID,AllowZoomTo,ColumnName,Created,CreatedBy,DDL_NoForeignKey,EntityType,FieldLength,IsActive,IsAdvancedText,IsAllowLogging,IsAlwaysUpdateable,IsAutocomplete,IsCalculated,IsDimension,IsDLMPartitionBoundary,IsEncrypted,IsForceIncludeInGeneratedModel,IsGenericZoomKeyColumn,IsGenericZoomOrigin,IsIdentifier,IsKey,IsLazyLoading,IsMandatory,IsParent,IsRangeFilter,IsSelectionColumn,IsShowFilterIncrementButtons,IsStaleable,IsSyncDatabase,IsTranslated,IsUpdateable,IsUseDocSequence,Name,SelectionColumnSeqNo,SeqNo,Updated,UpdatedBy,Version) VALUES (0,560054,544047,0,10,540965,'N','CustomerCode',TO_TIMESTAMP('2018-05-09 13:19:52','YYYY-MM-DD HH24:MI:SS'),100,'N','de.metas.shipper.gateway.derkurier',250,'Y','N','Y','N','Y','N','N','N','N','N','N','N','N','N','N','N','N','N','N','N','N','N','N','Y','N','Auftraggeber',0,0,TO_TIMESTAMP('2018-05-09 13:19:52','YYYY-MM-DD HH24:MI:SS'),100,0)
```
[the remark this answers is not in the document]
Comment (metas-rc) [reply de.metas.shipper.gateway.derkurier/src/main/sql/postgresql/system/99-de.metas.shipper.gateway.derkurier/5493040_sys_gh4034_DerKurierShipperConfig_CustomerCode_And_CollectorCode.sql]: Done

File: de.metas.shipper.gateway.spi/src/main/java/de/metas/shipper/gateway/spi/model/DeliveryOrder.java [hunk 4]
```diff
@@ -75,6 +75,12 @@
 
 	@NonNull
 	private ServiceType serviceType;
+	
+	@Nullable
+	private String collectorCode;
```
[the remark this answers is not in the document]
Comment (metas-rc) [reply de.metas.shipper.gateway.spi/src/main/java/de/metas/shipper/gateway/spi/model/DeliveryOrder.java]: Done

**Comment by metas-ts** [comment]
Hi @metas-rc , pls check out the build: https://jenkins.metasfresh.com/job/metasfresh-dist/job/master/10148/console
afaiu there is a null related problem in an SQL file. suggestion: set them to `00`

**Pull request merged**